### DISCUSSION TITLE
Creator notification

### DISCUSSION THREAD

**Discussion by norschu** [DNNCommunity/Dnn.CommunityForums#216] [Q&A] [answered]
Hello,
We have productively installed version 7.0.1. When someone creates a new post, all associated users are notified by email except the creator. Is there a way that the creator is also automatically notified?

**Comment by Timo-Breumelhof** [comment]
I guess you have the "Auto Subscriptions" option switched on but that only looks at the users "in the thread".
The Admin can simply subscribe to the forum and get notified that way (that's what I use)

**Comment by norschu** [comment]
Thank you for your prompt reply. We already use both options to subscribe. It is working. Our problem is that users who created a message themselves do not receive an email about it. We recently upgraded DNN from 7.42 to 9.10.2 and switched from core forums to active forums. In the core forum, the message creators still received the e-mail notifications themselves. Unfortunately not anymore. Can we set it up so that a forum post creator receives an automatic email for their own forum post?

**Reply by Timo-Breumelhof** [reply]
Ah I didn't read very well, Sorry about that.

**Comment by johnhenley** [comment, marked as answer]
> Hello,
> 
> We have productively installed version 7.0.1. When someone creates a new post, all associated users are notified by email except the creator. Is there a way that the creator is also automatically notified?

Hi @norschu thanks for using the forums and for your inquiry. I can confirm that there is logic that skips sending a subscription notification to the creator of the posting. 

We have options:
1. Leave it this way (not good for @norschu and perhaps others but keeping it working as expected). 
2. Remove this logic (which changes a current feature)
3. Make it a module setting
4. Make it a forum/forum group setting 
5. A user preference per forum/topic  "send or don't send me notifications when I post to this forum or topic"
6. Both 4 (as default) and 5 (as override). 

Community: PLEASE vote your preference. The good news is we have an open pull request #207 that touches this code so we might be able to address this quickly in an upcoming release.

**Reply by norschu** [reply]
Many thanks, option 4 would be the best for us. 🙏

**Reply by Timo-Breumelhof** [reply]
I think 4. would be best too.
`<edit> changed my mind: 2.</edit>`

**Reply by Timo-Breumelhof** [reply]
So that would be: "Auto Subscribe Creator"? 
So a user still has the option to unsubscribe?

**Reply by johnhenley** [reply]
> So that would be: "Auto Subscribe Creator"? 
> 
> So a user still has the option to unsubscribe?

This has nothing to do with automatic subscriptions. If a user posts a new topic (and is subscribed to the forum either specifically or as an automatic subscriber) or replies to an existing topic (whether subscribed to the topic or the forum either specifically or as an automatic subscriber), the user posting does not receive the notification email. 

There is literally an if statement that excludes the poster when building the subscriber list for a notification. 

That is why option 2 is easy, but would be a breaking change for existing installations. 

Option 4 is most difficult but ideally the best. It would add an option "include posting user when sending notifications" under forum and forum group features. This would  be set to NO for existing installations (to avoid breaking change), set to YES for new installations (maybe--we can discuss that option). And @norschu would set it to YES to achieve desired behavior.

**Reply by Timo-Breumelhof** [reply]
Ok I understand.
I would simply remove the statement as it's counter intuitive and makes no sense to exclude someone who has subscribed to their own post, right? 
No need for an option IMO. So I change my vote to 2. :-)

**Reply by johnhenley** [reply]
> Ok I understand.
> 
> I would simply remove the statement as it's counter intuitive and makes no sense to exclude someone who has subscribed to their own post, right? 
> 
> No need for an option IMO. So I change my vote to 2. :-)

Not necessarily that they are subscribed to their own post. If they are a forum subscriber, and reply to an existing topic or create a new topic, they currently don't get a notification that they posted (which to some people makes perfect sense but to others they might want a record of it). It almost begs for an option 5 :) a user preference per forum "send or don't send me notifications when I post to this forum or topic"

**Reply by Timo-Breumelhof** [reply]
hm, right.
5. then ;-)

**Reply by johnhenley** [reply]
> hm, right.
> 
> 5. then ;-)

Ha! Or maybe 4+5=6. A default option from the forum/forum group with a user-specific override per forum. This is becoming what we call a "creature feature".

**Comment by johnhenley** [comment]
@norschu Have you checked out latest release? We added this (#1220) in 08.02.00 release.
